frontend/pages/policies/ManagePoliciesPage/components/PoliciesTable/PoliciesTableConfig.tsx

```diff
@@ -283,34 +303,59 @@ const generateTableHeaders = (
     },
   ];
 
-  if (tableType !== "inheritedPolicies") {
-    if (!canAddOrDeletePolicy) {
-      return tableHeaders;
-    }
-
-    tableHeaders.unshift({
-      id: "selection",
-      Header: (cellProps: IHeaderProps) => {
-        const props = cellProps.getToggleAllRowsSelectedProps();
-        const checkboxProps = {
-          value: props.checked,
-          indeterminate: props.indeterminate,
-          onChange: () => cellProps.toggleAllRowsSelected(),
-        };
-        return <Checkbox {...checkboxProps} />;
-      },
-      Cell: (cellProps: ICellProps): JSX.Element => {
-        const props = cellProps.row.getToggleRowSelectedProps();
-        const checkboxProps = {
-          value: props.checked,
-          onChange: () => cellProps.row.toggleRowSelected(),
-        };
-        return <Checkbox {...checkboxProps} />;
-      },
-      disableHidden: true,
-    });
+  if (!hasPermissionAndPoliciesToDelete) {
+    return tableHeaders;
   }
 
+  tableHeaders.unshift({
+    id: "selection",
+    Header: (headerProps: any) => {
+      // When viewing team policies select all checkbox accounts for not selecting inherited policies
+      const teamCheckboxProps = configUtils.getConditionalSelectHeaderCheckboxProps(
```

Comment:
I found that [by comparing a query's team_id with the currently selected team_id, I could just use a single checkbox props](https://github.com/fleetdm/fleet/pull/18596/files#diff-9886cf37eb4b9e8c125e15e62c253673f9b2ba6eed1782ad27d0e3b8bd882a28R264) instead of one for each of these cases. Not sure if that's different for policies here because of pagination differences but maybe worth considering?